syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.registerallocationinference";
option java_outer_classname = "RegisterAllocationInferenceProto";
option objc_class_prefix = "REG";

package registerallocationinference;

// General Register Allocation Service 
service RegisterAllocationInference{

// RPC function to send Inference graph data and receive color data for codegen
  rpc getColouring(GraphList) returns (ColorData) {}
 
}

message GraphList {
  bytes payload=1;
}

message ColorData {
  bytes payload=2;
}
